pkg/controller/ps/crash_recovery.go

```diff
@@ -0,0 +1,112 @@
+package ps
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/clientcmd"
+	"github.com/percona/percona-server-mysql-operator/pkg/k8s"
+	"github.com/percona/percona-server-mysql-operator/pkg/mysql"
+	"github.com/percona/percona-server-mysql-operator/pkg/mysqlsh"
+)
+
+func (r *PerconaServerMySQLReconciler) reconcileFullClusterCrash(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
+	log := logf.FromContext(ctx).WithName("Crash recovery")
+
+	cli, err := clientcmd.NewClient()
+	if err != nil {
+		return err
+	}
+
+	pods, err := k8s.PodsByLabels(ctx, r.Client, mysql.MatchLabels(cr))
+	if err != nil {
+		return errors.Wrap(err, "get pods")
+	}
+
+	if len(pods) < int(cr.MySQLSpec().Size) {
+		return nil
+	}
+
+	// we need every pod to be ready to reboot
+	for _, pod := range pods {
+		if !k8s.IsPodReady(pod) {
+			return nil
+		}
+	}
+
+	operatorPass, err := k8s.UserPassword(ctx, r.Client, cr, apiv1alpha1.UserOperator)
+	if err != nil {
+		return errors.Wrap(err, "get operator password")
+	}
+
+	var outb, errb bytes.Buffer
+	cmd := []string{"/bin/bash", "-c", "cat /var/lib/mysql/full-cluster-crash"}
+
+	for _, pod := range pods {
+		err = cli.Exec(ctx, &pod, "mysql", cmd, nil, &outb, &errb, false)
+		if err != nil {
+			if strings.Contains(errb.String(), "No such file or directory") {
+				continue
+			}
+			return errors.Wrapf(err, "run %s, stdout: %s, stderr: %s", cmd, outb.String(), errb.String())
+		}
+
+		log.Info("Pod is waiting for recovery", "pod", pod.Name, "gtidExecuted", outb.String())
+
+		podFQDN := fmt.Sprintf("%s.%s.%s", pod.Name, mysql.ServiceName(cr), cr.Namespace)
+		podUri := fmt.Sprintf("%s:%s@%s", apiv1alpha1.UserOperator, operatorPass, podFQDN)
+
+		mysh, err := mysqlsh.NewWithExec(&pod, podUri)
+		if err != nil {
+			return err
+		}
+
+		err = mysh.RebootClusterFromCompleteOutageWithExec(ctx, cr.InnoDBClusterName())
+		if err == nil {
+			log.Info("Cluster was successfully rebooted")
+			r.Recorder.Event(cr, "Normal", "FullClusterCrashRecovered", "Cluster recovered from full cluster crash")
```

Comment:
Maybe first we should `cleanupFullClusterCrashFile` before recording this event because if something fails while cleaning up we would still be in full cluster crash "mode" because the fullClusterCrash file will be present on some pods. It could potentially be misleading.

Reply:
I get what you're saying but at this point we already rebooted the cluster. So cluster is recovered from full cluster crash. @hors what do you think?

Reply:
Now you got me thinking: do we need some check that the cluster is rebooted even though full-cluster-crash could be present on some pod? We can do `RebootClusterFromCompleteOutageWithExec` with success, but fail at cleaning up these files, then we would do `RebootClusterFromCompleteOutageWithExec` again next time we do `reconcileFullClusterCrash`?

Reply:
yes, that's correct. maybe we need to check if the file exists **and** something in the cluster status. if operator detects cluster is fine but these files exist, it can just delete files. I'll check if we can do something like this.

Reply:
Exactly that! Not sure if is it possible (or good practice), but could we reuse `FullClusterCrashRecovered` event for this check? Like, we do `FullClusterCrashRecovered` and then set this event (then do a cleanup). And to do a reboot, we check this event and if the files are present.

Reply:
if we can't detect the actual state from cluster, we can have a check like that but I don't think using events is a good practice. Having a condition would be better.

Reply:
> don't think using events is a good practice
I thought so. 

> Having a condition would be better.
Yes, definitely.